classes/User/User.mjs

```diff
@@ -18,6 +18,26 @@ export default class User {
     return this
   }
 
+  async updateProfile(data) {
+    if(!data) throw new Error("No data provided")
+
+    const existingUser = await database.findOne({ email: data.email }, "users")
+    if (existingUser && existingUser.email === data.email) {
+      throw new Error(`User with email ${data.email} already exists`)
+    }
+
+    const existingDisplayName = await database.findOne({ profile: { displayName: data.name } }, "users")
+    if (existingDisplayName && existingDisplayName.profile.displayName === data.name) {
+      throw new Error(`User with name ${data.name} already exists`)
+    }
+    
+    this.data = await (this.data ?? this.#loadFromDB().then(u => u.data))
+    this.data.email = data.email
+    this.data.profile.displayName= data.name
```

Comment:
ooo. I don't like this part.
Whatever is sent in as the body should be the content of the `profile` property (sanitized). We can publish an API that says we plan to surface publicly special things like "links" or "displayName" or "identities" or "homepage"